source/go-perun/app_tutorial/app.rst

```diff
@@ -0,0 +1,398 @@
+App
+===
+We discussed the basic blocks needed for creating a channel in the :ref:`payment channel example <payment_tutorial_intro>`.
+Additional channel parameters are required to define an app:
+An ``app`` field stores the address of an additional on-chain app contract and a ``data`` field that is added to the channel's state to insert arbitrary data for describing app states.
+
+We will implement two components in this section:
+
+#. An on-chain contract implementing the `Perun's Ethereum App interface <https://github.com/perun-network/perun-eth-contracts/blob/main/contracts/App.sol>`_. Most notably, the interface requires a ``validTransition`` function to validate the transition between app states to make the app logic enforceable without full consensus.
```

Comment:
- I like the ordering here. First introduce the on-chain component. Then the off-chain component. **Can you use the same order in the text?** (I started already by simply moving up the part `On-chain component`.)

- Please place all parts belonging to the off-chain component within the section `Off-chain component`. (At the moment, `Data` and `Utilities` have their own section at the same section level. If you feel all of this is too much for one page, you can also move `Off-chain component` and `On-chain component` to separate files.)

Reply:
I restructured the app section. Feel free to comment on the change.